pypesto/ensemble/ensemble.py

```diff
@@ -192,14 +205,31 @@ def _stack_outputs_sensi(ic: int):
             # stack into one numpy array
             return np.stack(output_sensi_list, axis=-1)
 
-        def _compute_summary(tmp_array, percentiles_list):
+        def _stack_weights(ic: int):
+            """
+            Group weights for different parameter vectors of one ensemble
+            together, if they belong to the same simulation condition, and
+            stacks them in one array
+            """
+            # Were outputs computed
+            if self.prediction_results[0].conditions[ic].output is None:
+                return None
+            # stack predictions
+            output_list = [prediction.conditions[ic].output[AMICI_LLH]
+                           for prediction in self.prediction_results]
```

Comment:
Hm... I have the impression that this will be problematic. By default, a PredictionConditionResult has a member "output", which is supposed to be an ndarray. You're making a dict out of it, which might brake quite some dependencies...
I would rather suggest you add a new member "output_weight" to the PredictionConditionResult. This way, you can just use this new member here, and an additional field will not affect the functioning of old script, hence not destroy backwards compatibility.